Net/SMTP.php

```diff
@@ -572,7 +572,17 @@ public function auth($uid, $pwd , $method = '', $tls = true, $authz = '')
             if (PEAR::isError($result = $this->parseResponse(220))) {
                 return $result;
             }
-            if (PEAR::isError($result = $this->socket->enableCrypto(true, STREAM_CRYPTO_METHOD_TLS_CLIENT))) {
+            if (isset($this->socket_options['ssl']['crypto_method'])) {
+                $crypto_method = $this->socket_options['ssl']['crypto_method'];
+            } else {
+                /* STREAM_CRYPTO_METHOD_TLS_ANY_CLIENT constant does not exist
+                 * and STREAM_CRYPTO_METHOD_SSLv23_CLIENT constant is
+                 * inconsistent across PHP versions. */
+                $crypto_method = STREAM_CRYPTO_METHOD_TLS_CLIENT
+                                 | @STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT
+                                 | @STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT;
```

Comment:
Another way to write this in a portable way would be

                $crypto_method = (STREAM_CRYPTO_METHOD_SSLv23_CLIENT
                                  & ~STREAM_CRYPTO_METHOD_SSLv2_CLIENT
                                  & ~STREAM_CRYPTO_METHOD_SSLv3_CLIENT)
                                 | STREAM_CRYPTO_METHOD_TLS_CLIENT;

but that seems odd to me.